proptest-state-machine/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "proptest-state-machine"
+version = "0.1.0"
+authors = ["Tomáš Zemanovič"]
+license = "MIT/Apache-2.0"
+edition = "2018"
+repository = "https://github.com/proptest-rs/proptest"
+homepage = "https://proptest-rs.github.io/proptest/proptest/state-machine.html"
+documentation = "https://docs.rs/proptest-state-machine/latest/proptest-state-machine/"
+keywords = ["property", "testing", "quickcheck", "fuzz", "state-machine"]
+categories = ["development-tools::testing"]
+description = """
+State machine based testing support for proptest.
+"""
+
+[dependencies]
+proptest = { version = "1.0.0", path = "../proptest" }
```

Comment:
we can't publish to crates with path dependencies:

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-path-dependencies

i think we'll also need to depend on a yet-to-be-released version "1.2.0" which has the changes to make the necessary symbols public.

Reply:
Thank you for much for your help and the thorough review! 

I think the docs say that we can't publish with a dependency that *only* has a path, but when there's also a `version` field it will use that for crates.io. You're right that the version needs to be bumped for this to work though. Let me add the changelogs, squash the fixup commits and rebase. I think we'll need to do the 1.2.0 release before publishing the state machine crate.

Reply:
The bump to 1.2.0 broke the CI for now. The last run before that was successful: https://github.com/proptest-rs/proptest/actions/runs/4870392917/jobs/8686088563. This is the diff: https://github.com/proptest-rs/proptest/compare/0e8a0c004957cb0ea41eb0072c04f26e594f70af..504393b33e4e306e7dc7018aa841e43d9de058b9